packages/core/admin/admin/src/components/FormInputs/Boolean.tsx

```diff
@@ -8,18 +9,23 @@ import { useField } from '../Form';
 
 import { InputProps } from './types';
 
-const BooleanInput = forwardRef<HTMLInputElement, InputProps>(
+const BooleanInput = React.forwardRef<HTMLInputElement, InputProps>(
   ({ name, required, label, hint, labelAction, ...props }, ref) => {
     const { formatMessage } = useIntl();
     const field = useField<boolean | null>(name);
     const fieldRef = useFocusInputField<HTMLInputElement>(name);
+    const [{ query }] = useQueryParams<{ plugins?: { i18n?: { locale?: string } } }>();
 
     const composedRefs = useComposedRefs(ref, fieldRef);
 
+    // Create a key that changes when locale changes to force re-render
+    const localeKey = query?.plugins?.i18n?.locale || 'default';
```

Comment:
Do these changes work correctly on a collection type that is not internationalised?